src/flamegraph/merge.rs

```diff
@@ -83,34 +110,30 @@ where
     let mut last = "";
     let mut tmp = Default::default();
     let mut frames = Default::default();
+    let mut delta = None;
+    let mut delta_max = 1;
     for line in lines {
         let mut line = line.trim();
         if line.is_empty() {
             continue;
         }
 
-        let nsamples = if let Some(samplesi) = line.rfind(' ') {
-            let mut samples = &line[(samplesi + 1)..];
-            // strip fractional part (if any);
-            // foobar 1.klwdjlakdj
-            if let Some(doti) = samples.find('.') {
-                samples = &samples[..doti];
-            }
-            match samples.parse::<usize>() {
-                Ok(nsamples) => {
-                    // remove nsamples part we just parsed from line
-                    line = line[..samplesi].trim_end();
-                    // give out the sample count
-                    nsamples
-                }
-                Err(_) => {
-                    ignored += 1;
-                    continue;
+        // Usually there will only be one samples column at the end of a line,
+        // but for differentials there will be two. In the case there are two
+        // we compute the delta between them and use the second one as as the
+        // number of samples for this frame.
+        let nsamples = match (parse_nsamples(&mut line), parse_nsamples(&mut line)) {
```

Comment:
I think this would be more readable as:
```rust
let nsamples = if let Some(s2) = parse_nsamples(&mut line) {
    let nsamples1 = parse_nsamples(&mut line);
    // ...
} else {
    ignored += 1;
    continue;
}
```